crates/gossip/src/gossip.rs

```diff
@@ -1115,4 +1122,46 @@ mod test {
             .unwrap();
         assert_eq!(7, all_ops.len());
     }
+
+    #[tokio::test]
+    async fn check_summary_after_round() {
+        enable_tracing();
+
+        let space = TEST_SPACE_ID;
+        let factory =
+            TestGossipFactory::create(space.clone(), true, None).await;
+        let harness_1 = factory.new_instance().await;
+        harness_1.join_local_agent(DhtArc::FULL).await;
+
+        let harness_2 = factory.new_instance().await;
+        let local_agent_2 = harness_2.join_local_agent(DhtArc::FULL).await;
+        let op_2 = MemoryOp::new(Timestamp::now(), vec![2; 128]);
+        let op_id_2 = op_2.compute_op_id();
+        harness_2
+            .op_store
+            .process_incoming_ops(vec![op_2.clone().into()])
```

Comment:
Not for this PR, but after you made me aware of it, I think it'd be better to have a test function to directly add ops to the store, instead of using this method.

Reply:
Well this is not what a host would do when adding ops to the store. It's called when new ops are coming in and has side-effects. In general it seems better to have one method for ops coming in through gossip and one for the host writing new ops to the store.

Reply:
Right, fair. So it will have the effect of storing the op in the memory op store and also immediately make it "integrated". We should probably be able to test this both ways, like this where it's convenient to make data available right away and also where it's not visible yet.